Detectors/TPC/reconstruction/src/ClustererTask.cxx

```diff
@@ -74,26 +76,38 @@ InitStatus ClustererTask::Init()
     return kERROR;
   }
 
+  mDigitMCTruthArray = mgr->InitObjectAs<decltype(mDigitMCTruthArray)>("TPCDigitMCTruth");
+  if( !mDigitMCTruthArray ) {
+    LOG(ERROR) << "TPC MC Truth not registered in the FairRootManager. Exiting ..." << FairLogger::endl;
+    return kERROR;
+  }
+
   if (mBoxClustererEnable) {
-    
+
     // Create and register output container
-    mClustersArray = new std::vector<o2::TPC::BoxCluster>;
+    mClustersArray = new std::vector<o2::TPC::Cluster>;
     mgr->RegisterAny("TPCCluster", mClustersArray, kTRUE);
 
+    // Register MC Truth output container
+    mClustersMCTruthArray = std::make_unique<MCLabelContainer>();
+    mgr->Register("TPCClusterMCTruth", "TPC", mClustersMCTruthArray.get(), kTRUE);
+
     // create clusterer and pass output pointer
-    mBoxClusterer = new BoxClusterer(mClustersArray);
-    mBoxClusterer->Init();
+    mBoxClusterer = std::make_unique<BoxClusterer>(mClustersArray);
   }
 
   if (mHwClustererEnable) {
     // Register output container
-    mHwClustersArray = new std::vector<o2::TPC::HwCluster>;
+    mHwClustersArray = new std::vector<o2::TPC::Cluster>;
     mgr->RegisterAny("TPCClusterHW", mHwClustersArray, kTRUE);
 
+    // Register MC Truth output container
+    mHwClustersMCTruthArray = std::make_unique<MCLabelContainer>();
+    mgr->Register("TPCClusterHWMCTruth", "TPC", mHwClustersMCTruthArray.get(), kTRUE);
```

Comment:
a tiny thing here: If we register the branch like this, we have the well known problem that the TBrowser cannot draw from the branch (see ROOT issue: https://sft.its.cern.ch/jira/browse/ROOT-8993) 

According to the ROOT team, the correct way to do it (in this case) is to add a trailing dot to the branch name like this
`TPCClusterHWMCTruth.`; I verified that this work but it needs adjustment also in the macro reading the clusters.

Shall we do it for the sake of usabibility? I can add this fix and merge it quickly.

Reply:
Yes, this sounds good. For me it is fine if you do this.
Please go ahead